app/pages/components/Creators.tsx

```diff
@@ -0,0 +1,179 @@
+      const provider = new anchor.Provider(connection, connectedWallet, OPTS);
+
+      return new anchor.Program<NftClub>(
+        IDL as unknown as NftClub,
+        PROGRAM_ID,
+        provider
+      );
+    }
+    return null;
+  }, [connectedWallet]);
+
+  const subscribeToCreator = useCallback(
+    async (
+      program: anchor.Program<NftClub>,
+      creatorPubKey: anchor.web3.PublicKey,
+      wallet: AnchorWallet,
+      creatorSolKey: anchor.web3.PublicKey
+    ) => {
+      setIsLoading(true);
+      const subscriptionSeeds = [
+        creatorPubKey.toBuffer(),
+        wallet.publicKey.toBuffer(),
+        anchor.utils.bytes.utf8.encode('subscription'),
+      ];
+      const [subscriptionPubKey] =
+        await anchor.web3.PublicKey.findProgramAddress(
+          subscriptionSeeds,
+          program.programId
+        );
+
+      await program.rpc.createSubscription({
+        accounts: {
+          subscription: subscriptionPubKey.toBase58(),
+          creator: creatorPubKey.toBase58(),
+          creatorSolAccount: creatorSolKey.toBase58(),
+          user: wallet.publicKey.toBase58(),
+          systemProgram: anchor.web3.SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+        },
+      });
+      setIsLoading(false);
+    },
+    []
+  );
+
+  const updateSubscription = useCallback(
+    async (
+      program: anchor.Program<NftClub>,
+      creatorPubKey: anchor.web3.PublicKey,
+      wallet: AnchorWallet,
+      creatorSolKey: anchor.web3.PublicKey
+    ) => {
+      setIsLoading(true);
+      const subscriptionSeeds = [
+        creatorPubKey.toBuffer(),
+        wallet.publicKey.toBuffer(),
+        anchor.utils.bytes.utf8.encode('subscription'),
+      ];
+      const [subscriptionPubKey] =
+        await anchor.web3.PublicKey.findProgramAddress(
+          subscriptionSeeds,
+          program.programId
+        );
+
+      await program.rpc.updateSubscription({
+        accounts: {
+          subscription: subscriptionPubKey.toBase58(),
+          creator: creatorPubKey.toBase58(),
+          creatorSolAccount: creatorSolKey.toBase58(),
+          user: wallet.publicKey.toBase58(),
+          systemProgram: anchor.web3.SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+        },
+      });
+      setIsLoading(false);
+    },
+    []
+  );
+
+  if (isLoading) {
+    return <progress className="progress w-56 place-content-center"></progress>;
+  }
+
+  return (
+    <div className="prose max-w-none lg:prose-xl">
+      <h1 className="px-10">Creators On This Network</h1>
+      <div className="grid grid-cols-1 px-10 py-5 md:grid-cols-3">
+        {creators.map((creator, key) => {
+          const creatorPubKey = creator.publicKey.toBase58();
+          const subscription = isSubscribed[creatorPubKey];
+          const subscriptionExpired = subscription
+            ? subscription.account.expireTimestamp.toNumber() < Date.now()
+            : true;
+          return (
```

Comment:
I would suggest making a separate Component for this. Not major, but best practices would be to minimize the amount of JS you're writing in the JSX. `.map` is fine and conditional rendering is fine, but pretty much everything else can be pulled out

Reply:
Makes sense